### PR TITLE
feat: Standardize error message format in Rust code

### PR DESCRIPTION
**Pull request by TheDataScientistNL**
This code change updates all error messages in the rust part of polars package; see #10421

### CODE REVIEW

**Comment by TheDataScientistNL** [comment]
@stinodego I have added many Rust error standardization corrections. Could you have a look? Not sure why the ubuntu 3.11 python tests fail. I have not, as far as I can see, changed anything that would result in the failing of a test with a `null` vs 0, and hesitant to push a fix....

**Comment by stinodego** [comment]
That's an unrelated bug caught by our parametric tests - don't worry about it! I'll review this soon.

**Comment by orlp** [comment]
This is a nitpick, but could you remove the periods from the end of error messages?

**Comment by stinodego** [comment]
> This is a nitpick, but could you remove the periods from the end of error messages?

Most don't have a period, right? The help text should have proper punctuation though. See https://github.com/pola-rs/polars/issues/10421

**Comment by ritchie46** [comment]
The single line messages shouldn't right?

**Comment by stinodego** [comment]
> The single line messages shouldn't right?

Correct. From the linked issue:

> * The error message starts with a single sentence, without capitalization or trailing period.
> * If further elaboration is required, two newlines are added before adding further text with proper capitalization and punctuation.

File: crates/polars-arrow/src/array/list/mod.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ impl<O: Offset> ListArray<O> {
         let child_data_type = Self::try_get_child(&data_type)?.data_type();
         let values_data_type = values.data_type();
         if child_data_type != values_data_type {
-            polars_bail!(ComputeError: "ListArray's child's DataType must match. However, the expected DataType is {child_data_type:?} while it got {values_data_type:?}.");
+            polars_bail!(ComputeError: "ListArray's child's DataType must match, yet the expected DataType is {child_data_type:?} while it got {values_data_type:?}");
```
Comment (stinodego) [inline crates/polars-arrow/src/array/list/mod.rs:55]: This is too long for a single line error message. It should be broken up into a summary and a help text.

File: crates/polars-arrow/src/array/map/mod.rs [hunk 2]
```diff
@@ -131,7 +131,7 @@ impl MapArray {
         if let DataType::Map(field, _) = data_type.to_logical_type() {
             Ok(field.as_ref())
         } else {
-            polars_bail!(ComputeError: "The data_type's logical type must be DataType::Map")
+            polars_bail!(ComputeError: "the data_type's logical type must be DataType::Map")
```
Comment (stinodego) [inline crates/polars-arrow/src/array/map/mod.rs:134]: Code snippets must be in backticks.

```suggestion
            polars_bail!(ComputeError: "the data_type's logical type must be `DataType::Map`")
```

**Review by stinodego — CHANGES_REQUESTED** [review]
A lot of improvements, but a lot of messages are now less clear because you stitched multi-sentence error messages together, instead of breaking them up into a short summary and a help text. Also, a lot of code snippets are still not in backticks.

This PR is too big for me to sit down and go through every example and correct the messages that I feel are regressions. Please break this up into multiple PRs, let's say one per crate, and really take care that the messages are high quality. Then we can actually get some of this work merged!

**Comment by TheDataScientistNL** [comment]
> A lot of improvements, but a lot of messages are now less clear because you stitched multi-sentence error messages together, instead of breaking them up into a short summary and a help text. Also, a lot of code snippets are still not in backticks.
> 
> This PR is too big for me to sit down and go through every example and correct the messages that I feel are regressions. Please break this up into multiple PRs, let's say one per crate, and really take care that the messages are high quality. Then we can actually get some of this work merged!

Will do.

**Pull request closed without merging**